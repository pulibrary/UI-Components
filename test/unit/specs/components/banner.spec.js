import { createLocalVue, mount } from "@vue/test-utils"
import Banner from "@/elements/Banner.vue"

// create an extended `Vue` constructor
const localVue = createLocalVue()

describe("Banner.vue", () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(Banner, {
      localVue,
      slots: {
        default: "Here's some info for you.",
      },
      stubs: ["wrapper"],
    })
  })

  it("should have an alert message", () => {
    const el = wrapper.find(".lux-banner")
    expect(el.text()).toBe("Here's some info for you.")
  })

  it("should have a button when dismissible", () => {
    wrapper.setProps({ dismissible: true })
    const button = wrapper.find("button")
    expect(wrapper.vm.dismissible).toBe(true)
    expect(button.is("button")).toBe(true)
  })

  it("should be dismissible on click", () => {
    wrapper.setProps({ dismissible: true })
    const button = wrapper.find("button")
    button.trigger("click")
    expect(wrapper.isEmpty()).toBe(true)
  })

  it("should hide banner when hideBanner() is called", () => {
    expect(wrapper.vm.show).toBe(true)
    wrapper.vm.hideBanner()
    expect(wrapper.vm.show).toBe(false)
  })

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
