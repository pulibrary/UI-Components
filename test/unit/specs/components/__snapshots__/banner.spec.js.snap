// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner.vue has the expected html structure 1`] = `
<div
  class="lux-banner"
  name="fade"
>
  <div
    class="lux-announcement open"
  >
    <wrapper
      fullwidth="true"
    >
      <!---->
       Here's some info for you.
    </wrapper>
  </div>
</div>
`;
